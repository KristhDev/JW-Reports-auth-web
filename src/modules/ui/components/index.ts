export { Button } from './Button';
export { EyeBtn } from './EyeBtn';
export { FormField } from './FormField';
export { Spinner } from './Spinner';
export { Title } from './Title';