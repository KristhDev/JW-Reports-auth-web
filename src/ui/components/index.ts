export { FormField } from './FormField';
export { Title } from './Title';