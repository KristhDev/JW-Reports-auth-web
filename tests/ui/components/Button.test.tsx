import { fireEvent, render, screen } from '@testing-library/react';

import { Button } from '../../../src/ui/components';

const btnText = 'Test text';
const onClickMock = vi.fn();

describe('Test in <Button /> component', () => {
    beforeEach(() => {
        render(
            <Button
                className="test-class"
                onClick={ onClickMock }
                type="button"
            >
                { btnText }
            </Button>
        );

        vi.clearAllMocks();
    });

    it('should to match the snapshot', () => {
        const { container } = render(
            <Button
                onClick={ onClickMock }
                type="button"
            >
                { btnText }
            </Button>
        );

        expect(container).toMatchSnapshot();
    });

    it('should render respective props', () => {
        const btn = screen.getByRole<HTMLButtonElement>('button');

        expect(btn.className).toContain('test-class');
        expect(btn.innerHTML).toBe(btnText);
        expect(btn.type).toBe('button');
    });

    it('should call onClick when clicked', () => {
        const btn = screen.getByRole<HTMLButtonElement>('button');
        fireEvent.click(btn);

        expect(onClickMock).toHaveBeenCalledTimes(1);
    });
});