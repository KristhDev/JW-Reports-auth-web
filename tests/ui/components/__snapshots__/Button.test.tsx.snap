// Vitest Snapshot v1

exports[`Test in <Button /> component > should to match the snapshot 1`] = `
<div>
  <button
    class="button undefined"
    type="button"
  >
    Test text
  </button>
</div>
`;
